src/service/osw-service.ts

```diff
@@ -127,7 +127,8 @@ class OswService implements IOswService {
       * @throws If there is an error processing the tagging request.
       */
     async calculateTagQualityMetric(tdei_dataset_id: string, tagFile: any, user_id: string): Promise<any> {
-
+        //Check dataset exists
+        await this.tdeiCoreServiceInstance.getDatasetDetailsById(tdei_dataset_id);
```

Comment:
why is this called here if its not used?

Reply:
Function throws 404 error when dataset id not found